tracer/src/Datadog.Tracer.Native/debugger_probes_instrumentation_requester.cpp

```diff
@@ -618,7 +617,7 @@ void DebuggerProbesInstrumentationRequester::InstrumentProbes(
         std::copy(revertRequests.begin(), revertRequests.end(), requests.begin());
         auto promise = std::make_shared<std::promise<void>>();
         std::future<void> future = promise->get_future();
-        m_debugger_rejit_preprocessor->EnqueueRequestRevert(requests, promise);
+        m_debugger_rejit_preprocessor->EnqueueRequestRejit(requests, promise);
```

Comment:
Same comment that I wrote regarding the fault-tolerant instrumentation: debugger's instrumentation rely on receiving the original bytecode.
Debugger's instrumentation in a nutshell: Each method that go through instrumentation, could have `X` amount of probes. When we instrument a method, we apply all the _currently active_ probes on the method. If a customer. using Datadog's UI, decides to delete a probe, we re-instrument the method and re-apply the _newly active probes_. To be able to do that, we have to `Revert` & `Rejit` the method, so we'll have the original bytecode, and reapply all the probes on that. If you won't call `Revert` (as suggested by this change, and also the change mentioned in the fault-tolerant code) - we might receive the instrumented version of the bytecode by the debugger rewriter and we won't be able to reapply the probes, effectively breaking our instrumentation entirely.
Does that make sense?

Reply:
It's the same as above. Rejit calls internally a Revert. Theoretically there is no need to call Revert explicitly, but if we prove it is not needed, we can add it. Besides, Debugger tests work with current implementation.

Reply:
From the docs it makes sense but I tend to disbelieve Microsoft when it comes to docs and explanations 😅 they tend to be inaccurate to say the least, especially around the CLR Profiling APIs.
I will check it myself technically to see how things work among the different .NET runtimes. If that's indeed the case, then I don't mind getting rid of `RequestRevert`.

Reply:
Great, thanks a lot. But, as I said, if finally is necessary to explicitly call Revert, it can be done with minimal modifications.

Reply:
Just as a note, we discussed via Zoom and decided to keep the `Revert` call for the good orders sake. This PR touches too much of a sensitive locations, thus wherever possible to keep the execution as it was is preferred. We don't know what we don't know.

Reply:
It's been implemented here a0f39b5fdf49a636c79820aea50f9477c65e1af2